src/WebJobs.Script.WebHost/Middleware/ExceptionMiddleware.cs

```diff
@@ -28,11 +28,18 @@ public async Task Invoke(HttpContext context)
             }
             catch (Exception ex)
             {
+                // Exceptions thrown by function code are handled/logged elsewhere.
+                // our goal here is to log exceptions coming from our own runtime
                 if (!(ex is FunctionInvocationException))
                 {
-                    // exceptions throw by function code are handled/logged elsewhere
-                    // our goal here is to log exceptions coming from our own runtime
-                    Logger.UnhandledHostError(_logger, ex);
+                    if (ex is TaskCanceledException)
```

Comment:
We we certain here that every instance of `TaskCancelledException` would be an invocation cancellation? There are other flows in the Invoke path that could lead to this being thrown, right? Are we trying to differentiate that in an y way, or is it accurate to interpret as an invocation cancellation?

We know when an invocation is cancelled. Would it be more accurate to have a specific exception that communicates that so there's no ambiguity? That could actually be a `FunctionInvocationException` derived type. Thoughts?